server/src/internalClusterTest/java/org/opensearch/index/ClusterMaxMergesAtOnceIT.java

```diff
@@ -96,11 +95,15 @@ public void testClusterLevelDefaultUpdatesMergePolicy() throws ExecutionExceptio
         assertEquals(20, ((OpenSearchTieredMergePolicy) indexService.getIndexSettings().getMergePolicy(true)).getMaxMergeAtOnce());
         assertEquals(20, ((OpenSearchTieredMergePolicy) secondIndexService.getIndexSettings().getMergePolicy(true)).getMaxMergeAtOnce());
 
+        int replicas = randomIntBetween(1, Math.max(1, internalCluster().numDataNodes() - 1));
```

Comment:
More for my own curiosity than anything -- how does this end up fixing the test?

By the default, the replica count would be 1, based on the implementation of `indexSettings()` on line 56. So I don't understand how setting the replica count to a random number between 1 and 1 (since it looks like `numDataNodes = 2`) changes anything.

I'm not particularly familiar with these tests, but I'm trying to understand more. Thanks!

Reply:
because this
 ```java
  
createIndex(
            indexName,
            Settings.builder().put(TieredMergePolicyProvider.INDEX_MERGE_POLICY_MAX_MERGE_AT_ONCE_SETTING.getKey(), 15).build()
        );
```
overrides  the settings in line 56 and ends up generating intermittently zero or one number of replica.

The solution is to force it to generate at least one replica. 
On the other hand, I deliberately chose the max number to be random rather than a strict one to add more randomness to the test in case _numDataNodes_ changes in the future.
Hope this makes sense.

Reply:
@andrross 
 
Yeah you're right that the test would go back flaky if the _numDataNodes_ changes in the future, but  I believe the same goes with your solution of merging the settings to keep _number_of_replicas_=1. Both of our solutions would fix the bug if _numDataNodes_ stays 2 but fail if the number is higher. So I see 2 options

- Stick with the setup settings (_number_of_replicas_=1) and _numDataNodes_=2 (use _prepareCreate_ to merge the settings).

- Add a a helper function that finds the right node hosting the index. This would make the tests pass whatever the number of replicas, shards or nodes.

What do you think?